src/diff.sh

```diff
@@ -1,98 +1,49 @@
-include "${KW_LIB_DIR}/lib/kwio.sh"
 include "${KW_LIB_DIR}/lib/kwlib.sh"
+include "${KW_LIB_DIR}/lib/kwio.sh"
 
-# Hash containing user options
-declare -gA diff_options
+declare -gA options_values
 
-function diff_manager()
+function diff_main()
 {
-  local files_paths=${*: -2}
-  local interactive
-  local target_1
-  local target_2
+  local target_a="$1"
+  local target_b="$2"
+  local flag
 
-  IFS=' ' read -r -a files <<< "$files_paths"
+  IFS=' ' read -r -a files <<< "$target_a $target_b"
   for file in "${files[@]}"; do
     if [[ "$file" =~ -h|--help ]]; then
       diff_help "$file"
       return 0
     fi
 
     if [[ ! -e "$file" ]]; then
-      complain "Invalid path: $file"
+      complain "Invalid path: ${file}"
       diff_help
       return 2 # ENOENT
     fi
   done
 
-  target_1="${files[0]}"
-  target_2="${files[1]}"
-
-  # Drop files
-  set -- "${@:1:$#-2}"
-
-  diff_parser_options "$@"
-  if [[ "$?" -gt 0 ]]; then
-    complain "Invalid option: ${diff_options['ERROR']}"
-    return 22
+  parse_diff_options "$@"
+  if [[ $? -gt 0 ]]; then
+    complain "Invalid option: ${options_values['ERROR']}"
+    diff_help
+    exit 22 # EINVAL
   fi
 
-  if [[ "${diff_options['HELP']}" == 1 ]]; then
-    diff_help "$@"
-    return 0
-  fi
-
-  interactive="${diff_options['INTERACTIVE']}"
-  test_mode="${diff_options['TEST_MODE']}"
+  interactive="${options_values['INTERACTIVE']}"
+  test_mode="${options_values['TEST_MODE']}"
 
   if [[ "$test_mode" == 'TEST_MODE' ]]; then
-    printf '%s\n' "$target_1 $target_2 $interactive"
+    printf '%s\n' "$target_a $target_b $interactive"
```

Comment:
Use one `%s` per variable, and separate those variables from the double-quotes.

Reply:
@aquilamacedo, considering the latest force-push, I think you forgot to also replace `'%s\n'` for `'%s %s %s\n'`